week3/homework/src/actions.js

```diff
@@ -0,0 +1,100 @@
+let fs = require('fs');
+
+function readTodo(request, response) {
+  fs.readFile('./todoList.json', 'utf8', (error, todoList) => {
+    const parsedList = JSON.parse(todoList);
+    if (error) {
+      if (error.code === 'EN0ENT') {
+        console.log('No data found');
+      } else {
+        console.error(error);
+        response.status(500).send('some problem.');
+      }
+    } else if (request.params.id < parsedList.length) {
+      const todoItem = parsedList[++request.params.id]['text'];
+      response.json(todoItem);
+    } else {
+      response.json('Please enter a valid id');
+    }
+  });
+}
+
+function add(request, response) {
+  fs.readFile('./todoList.json', 'utf8', (error, data) => {
+    if (error) {
+      console.log(error);
```

Comment:
You need always to respond to client; otherwise the request will keep hanging until it timeout